public/js/Filter.js

```diff
@@ -1,23 +1,26 @@
-function SampleFilter(uniqueRow, uniqueCol, uniquePlate, uniqueGene) {
+var _ = require('lodash');
+
+function SampleFilter(sampleData) {
```

Comment:
I've finally figured out why I have so much trouble reading js code... Docstrings! I've been massively spoiled by Python docstrings. Is there an equivalent in js, and if so, can you sprinkle them all over this PR? In particular, for each function I would like:

- description of what the function does
- inputs (including type)
- outputs
- side effects (e.g. here, "adds xyz fields to the SampleFilter object")